scripts/build-renode.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+
+if [ "`whoami`" = "root" ]
+then
+	echo "Running the script as root is not permitted"
+	exit 1
+fi
+
+CALLED=$_
+[[ "${BASH_SOURCE[0]}" != "${0}" ]] && SOURCED=1 || SOURCED=0
+
+SCRIPT_SRC=$(realpath ${BASH_SOURCE[0]})
+SCRIPT_DIR=$(dirname $SCRIPT_SRC)
+TOP_DIR=$(realpath $SCRIPT_DIR/..)
+
+if [ $SOURCED = 1 ]; then
+	echo "You must run this script, rather then try to source it."
+	echo "$SCRIPT_SRC"
+	exit 1
+fi
+
+if [ -z "$HDMI2USB_ENV" ]; then
+	echo "You appear to not be inside the HDMI2USB environment."
+	echo "Please enter environment with:"
+	echo "  source scripts/enter-env.sh"
+	exit 1
+fi
+
+# Imports TARGET, PLATFORM, CPU and TARGET_BUILD_DIR from Makefile
+eval $(make env)
+make info
+
+source $SCRIPT_DIR/settings.sh
+
+set -x
+set -e
+
+RENODE_BIN=${RENODE_BIN:-renode}
+if ! command -v $RENODE_BIN 2>&1 1>/dev/null; then
+	# Download prebuilt renode Release if none is currently installed
+
+	RENODE_PACKAGE=renode-${RENODE_VERSION}-1-x86_64.pkg.tar.xz
+	RENODE_URL=https://github.com/renode/renode/releases/download/v$RENODE_VERSION/$RENODE_PACKAGE
+	RENODE_LOCATION="$BUILD_DIR/renode"
+	RENODE_BIN=$RENODE_LOCATION/opt/renode/bin/Renode.exe
+
+	if [ ! -x $RENODE_BIN ]; then
+		mkdir -p $RENODE_LOCATION
+		(
+			cd $RENODE_LOCATION
+			wget $RENODE_URL
+			tar -xf $RENODE_PACKAGE
+		)
+		
+		chmod u+x $RENODE_BIN
+		echo "Renode downloaded and installed locally: $RENODE_BIN" 
+	fi
+fi
+
+case $CPU in
+	vexriscv | picorv32)
+		;;
+	*)
+		echo "CPU $CPU_TYPE isn't supported at the moment."
+		exit 1
+		;;
+esac
+
+LITEX_CONFIG_FILE="$TARGET_BUILD_DIR/test/csr.csv"
+if [ ! -f $LITEX_CONFIG_FILE ]; then
+	make firmware
+fi
+
+function parse_generated_header {
```

Comment:
This is trying to get the IP address right? I would suggest we use the following name...
```suggestion
function get_ethernet_ip_address {
```

This is probably useful in the qemu emulation too. We should do that there too?

Reply:
Before moving to a python script this function was used to extract different values defined in `mem.h`/`csr.h` files. Right now it's used in just one context, so it lost its universality - refactoring and renaming it seems like a good idea.

The whole tap-related fragment of script is a copy-paste from `build-qemu.sh`, so the two scripts shares a lot more code. I will extract it to some base file and source in both `build-renode.sh` and `build-qemu.sh`.

Reply:
After extracting the common function to `build-common.sh` it turns out that `parse_generated_header` function might be used in `build-qemu` in several places (in different context), so I kept it's name.